IdentityCore/src/cache/crypto/MSIDAssymetricKeyKeychainGenerator.m

```diff
@@ -131,60 +130,56 @@ - (MSIDAssymetricKeyPair *)readOrGenerateKeyPairForAttributes:(MSIDAssymetricKey
 - (MSIDAssymetricKeyPair *)readKeyPairForAttributes:(MSIDAssymetricKeyLookupAttributes *)attributes
                                              error:(NSError **)error
 {
-    if ([NSString msidIsStringNilOrBlank:attributes.privateKeyIdentifier]
-        || [NSString msidIsStringNilOrBlank:attributes.publicKeyIdentifier])
+    if ([NSString msidIsStringNilOrBlank:attributes.privateKeyIdentifier])
     {
         [self logAndFillError:@"Invalid key lookup attributes provided" status:-1 error:error];
         return nil;
     }
     
-    NSDictionary *privateKeyAttributes = [self keyAttributesWithQueryDictionary:[attributes privateKeyAttributes] keyTitle:@"private key" error:error];
-    
-    if (!privateKeyAttributes)
+    NSDictionary *privateKeyDict = [self keyAttributesWithQueryDictionary:[attributes privateKeyAttributes] error:error];
+    if (!privateKeyDict)
     {
         return nil;
     }
     
-    NSDictionary *publicKeyAttributes = [self keyAttributesWithQueryDictionary:[attributes publicKeyAttributes] keyTitle:@"public key" error:error];
-    
-    if (!publicKeyAttributes)
+    SecKeyRef privateKeyRef = (__bridge SecKeyRef)privateKeyDict[(__bridge id)kSecValueRef];
+    if (!privateKeyRef)
     {
         return nil;
     }
     
-    SecKeyRef privateKeyRef = (__bridge SecKeyRef)privateKeyAttributes[(__bridge id)kSecValueRef];
-    SecKeyRef publicKeyRef = (__bridge SecKeyRef)publicKeyAttributes[(__bridge id)kSecValueRef];
-    
-    if (!privateKeyRef || !publicKeyRef)
+    SecKeyRef publicKeyRef = SecKeyCopyPublicKey(privateKeyRef);
+    if (!publicKeyRef)
     {
-        [self logAndFillError:@"Invalid keychain attributes. No key ref returned" status:-1 error:error];
         return nil;
     }
     
+    NSDate *creationDate = [privateKeyDict objectForKey:(__bridge NSDate *)kSecAttrCreationDate];
+    
     MSIDAssymetricKeyPair *keypair = [[MSIDAssymetricKeyPair alloc] initWithPrivateKey:privateKeyRef
```

Comment:
Where is publicKeyRef getting released?

Reply:
It will get retained in the MSIDAssymetricKeyPair constructor and released in its dealloc method.